rust/otap-dataflow/crates/config/src/pipeline.rs

```diff
@@ -612,11 +622,207 @@ impl Default for PipelineConfigBuilder {
+            } else {
+                return Err(serde::de::Error::custom(
+                    "Expected either 'periodic' or 'pull' reader",
+                ));
+            }
+        } else {
+            return Err(serde::de::Error::custom(
+                "Invalid configuration of a metrics reader",
+            ));
+        }
+    }
+}
+
+/// Opentelemetry Metrics Perioidc Exporter configuration.
+#[derive(Debug, Clone, Serialize, JsonSchema)]
+pub enum MetricsPeriodicExporterConfig {
+    /// Console exporter that writes metrics to the console.
+    Console(ConsoleExporterConfig),
+    /// OTLP exporter that sends metrics using the OpenTelemetry Protocol.
+    Otlp,
+}
+
+/// Opentelemetry Console Exporter configuration.
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
+pub struct ConsoleExporterConfig {
+    /// The temporality preference for the console exporter.
+    #[serde(rename = "temporality")]
+    pub temporality: Option<String>,
+}
+
+impl<'de> Deserialize<'de> for MetricsPeriodicExporterConfig {
+    /// Custom deserialization to handle different exporter types.
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        use serde::de::{MapAccess, Visitor};
+        use std::fmt;
+        struct MetricsPeriodicExporterConfigVisitor;
+
+        impl<'de> Visitor<'de> for MetricsPeriodicExporterConfigVisitor {
+            type Value = MetricsPeriodicExporterConfig;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+                formatter.write_str("a map with either 'console' or 'otlp' key")
+            }
+
+            fn visit_map<M>(self, mut map: M) -> Result<Self::Value, M::Error>
+            where
+                M: MapAccess<'de>,
+            {
+                while let Some(key) = map.next_key::<String>()? {
+                    match key.as_str() {
+                        "console" => {
+                            let console_config: ConsoleExporterConfig = map.next_value()?;
+                            return Ok(MetricsPeriodicExporterConfig::Console(console_config));
+                        }
+                        "otlp" => {
+                            let _: () = map.next_value()?;
+                            return Ok(MetricsPeriodicExporterConfig::Otlp);
+                        }
+                        _ => {
+                            return Err(serde::de::Error::unknown_field(
+                                &key,
+                                &["console", "otlp"],
+                            ));
+                        }
+                    }
+                }
+                Err(serde::de::Error::custom(
+                    "Expected either 'console' or 'otlp' exporter",
+                ))
+            }
+        }
+
+        deserializer.deserialize_map(MetricsPeriodicExporterConfigVisitor)
+    }
+}
```

Comment:
nit - Is there plan to eventually use the [opentelemetry-config](https://github.com/open-telemetry/opentelemetry-rust-contrib/tree/main/opentelemetry-config) crate once it is published?

Reply:
we can publish it anytime. No need to wait for usual release cycle.